src/main/java/seedu/address/logic/parser/FilterByDateTimeParser.java

```diff
@@ -0,0 +1,119 @@
+package seedu.address.logic.parser;
+
+
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_COMMAND_FORMAT;
+import static seedu.address.logic.commands.FilterByDateCommand.ERROR_MESSAGE_INVALID_FORMAT;
+import static seedu.address.logic.commands.FilterByDateCommand.ERROR_MESSAGE_INVALID_TAG;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeParseException;
+import java.util.Arrays;
+import java.util.List;
+
+import seedu.address.logic.commands.FilterByDateCommand;
+import seedu.address.logic.commands.FindTaskCommand;
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.task.TaskBetweenDatesPredicate;
+/**
+ * Parses input arguments and creates a new FilterByDate Command
+ */
+public class FilterByDateTimeParser implements Parser<FilterByDateCommand> {
+
+    /**
+     * Parses the given {@code String} of arguments in the context of the FilterByDateTimeCommand
+     * and returns a FilterByDateCommand object for execution.
+     * @throws ParseException if the user input does not conform the expected format
+     */
+    public FilterByDateCommand parse(String args) throws ParseException {
+        String trimmedArgs = args.trim();
+        if (trimmedArgs.isEmpty()) {
+            throw new ParseException(
+                    String.format(MESSAGE_INVALID_COMMAND_FORMAT, FindTaskCommand.MESSAGE_USAGE));
+        }
+
+        return new FilterByDateCommand(new TaskBetweenDatesPredicate(inBetweenDates(trimmedArgs)));
+    }
+
+    /**
+     * Gets the before and after date from "filter d/22-08-2022 0800,23-08-2023 0800"
+     * @param dates dates to be seperated
+     * @return A list of before and after dates
+     */
+    public List<LocalDateTime> inBetweenDates(String dates) throws DateTimeParseException, ParseException {
+        // from "d/22-08-2022,23-08-2022" to ["d", "22-08-2022 0800", "23-08-2022 0800"]
+        String[] splitDates = dates.split("[/,]");
+        if (splitDates.length != 3) {
+            throw new ParseException(ERROR_MESSAGE_INVALID_TAG);
+        }
+
+        /*
+        trim() used to prevent whitespaces in " 23-08-2022 0800",
+        which will cause the String array to be {"", "23", "08", "2022", "0800"},
+        as we want only {"23", "08", "2022", "0800"}
+        */
+        String[] datetime1 = splitDates[1].trim().split("[ -]");
+        String[] date2 = splitDates[2].trim().split("[ -]");
+        // no time provided
+        if (datetime1.length == 3) {
+            return dayMonthYear(datetime1, date2);
+        } else {
+            return dayMonthYearTime(datetime1, date2);
+        }
+    }
+
+    private List<LocalDateTime> dayMonthYear(String[] date1, String[] date2) throws ParseException {
+        Integer[] date1Int = new Integer[3];
+        Integer[] date2Int = new Integer[3];
+        // date1Int = [dd, mm, yyyy, HH, MM]
+        for (int i = 0; i < date1.length; i++) {
+            date1Int[i] = Integer.parseInt(date1[i]);
```

Comment:
Potential Index out of Bounds Exception to be handled? Like if date1.length > 3 then doing date1Int[i] will cause problems